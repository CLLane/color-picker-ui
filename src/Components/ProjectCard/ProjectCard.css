.project-card__header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  border-bottom: solid 2px black;
  margin-bottom: 5px;
  padding-left: 15px;
  padding-right: 15px;
  height: 30px;
}
.project-card__article {
  border-radius: 8px;
  background: rgb(245, 245, 245, .9);
  margin-right: 10px;
  padding: 5px;
  margin-top: 10px;
  box-shadow: 4px 4px 10px black;
}

.project-edit__input {
  background: rgb(0,0,0,0);
  font-size: 20px;
  font-weight: bold;
  height: 25px;
}

.project-edit__input::placeholder {
  color: lightgrey;
}