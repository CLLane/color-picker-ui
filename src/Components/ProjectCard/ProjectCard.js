import React, { Component } from 'react';
import PaletteCard from '../PaletteCard/PaletteCard';
import editIcon from '../../Images/editIcon.svg';
import trashIcon from "../../Images/trashIcon.svg";
import saveIcon from '../../Images/saveIcon.svg';
import './ProjectCard.css';


export class ProjectCard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      nameInput: props.project.name,
      disabled: true,
      error: ''
    }
  }
  
  editName = () => {
    this.setState({ disabled: false });
  };

  saveName = async () => {
    const { project, updateProjectName } = this.props;
    const { nameInput } = this.state;
    const newProject = {
      ...project,
      name: nameInput
    };
    updateProjectName(newProject);
    this.setState({ disabled: true});
  }

  handleChange = (e) => {
    this.setState({ nameInput: e.target.value})
  }


  render() {
    const { project, palettes, trashPalette, trashProject, showPalette, updatePaletteName} = this.props;
    const { disabled, nameInput, error } = this.state;
    const paletteCards = palettes.map((palette, index) => {
      return <PaletteCard key={index} palette={palette} trashPalette={trashPalette} showPalette={showPalette} updatePaletteName={updatePaletteName} />
    });

    return (
      <article className='project-card__article'>
        <div className='project-card__header'>
          { error && <p>{error}</p> }
          { disabled && <img src={editIcon} alt='edit icon' onClick={this.editName}/> } 
          <input type='text' onChange={this.handleChange} disabled={disabled} value={nameInput}></input>
          { !disabled && <img src={saveIcon} alt='save icon' onClick={this.saveName}/> }
          <img src={trashIcon} alt='trashIcon' onClick={() => trashProject(project.id)}/>
        </div>
        { paletteCards }
      </article>
    ) 
  }
};

export default ProjectCard;