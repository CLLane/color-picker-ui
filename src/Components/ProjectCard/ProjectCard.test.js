import React from 'react';
import { shallow } from 'enzyme';
import ProjectCard from './ProjectCard';
import './ProjectCard.css';

describe('Project Card', () => {
  let wrapper;
  let mockProject;
  let mockPalette;
  let mockPalettes;
  let mockShowPalette;
  let mockUpdateProjectName;
  let mockUpdatePaletteName;
  let mockTrashPalette;
  let mockGrabPalette;
  let mockTrashProject;
  beforeEach(() => {
    mockPalettes = [{
      id: 1, 
      name: 'Cool Colors',
      project_id: '12345',
      colorOne: '#FFFFFF',
      colorTwo: '#FFFFFF',
      colorThree: '#FFFFFF',
      colorFour: '#FFFFFF',
      colorFive: '#FFFFFF'
    },
    ];
    mockPalette = {
      id: 1, 
      name: 'Cool Colors',
      project_id: '12345',
      colorOne: '#FFFFFF',
      colorTwo: '#FFFFFF',
      colorThree: '#FFFFFF',
      colorFour: '#FFFFFF',
      colorFive: '#FFFFFF'
    };
    mockProject = {
      id: 12345,
      name: 'Cool Project',
      user_id: 1
    };
    mockUpdateProjectName = jest.fn();
    mockUpdatePaletteName = jest.fn();
    mockTrashPalette = jest.fn();
    mockGrabPalette = jest.fn();
    mockShowPalette = jest.fn();
    mockUpdatePaletteName = jest.fn();
    wrapper = shallow(<ProjectCard 
      project={mockProject}
      palette={mockPalette}
      palettes={mockPalettes}
      trashPalette={mockTrashPalette}
      trashProject={mockTrashProject}
      grabPalette={mockGrabPalette}
      showPalette={mockShowPalette}
      updateProjectName={mockUpdateProjectName}
      updatePaletteName={mockUpdatePaletteName}
    />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot when the project name is edittable', () => {
    wrapper.setState({disabled: false});
    expect(wrapper).toMatchSnapshot();

  });
  
  it('should allow the user to edit a project', () => {
    wrapper.find('img').at(0).simulate('click');
    expect(wrapper.state('disabled')).toEqual(false);
  });

  it('should allow the user to edit the name of the project', () => {
    wrapper.setState({ nameInput: 'Bob', disabled: false });
    wrapper.find('img').at(0).simulate('click');
    expect(mockUpdateProjectName).toHaveBeenCalledWith({...mockProject, name: 'Bob'})
  });


  it('should update the project name with what a user entered', () => {
    const mockEvent = {
      target: {
        value: 'amazing palette'
      }
    };
    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state('nameInput')).toEqual('amazing palette');
  });


});