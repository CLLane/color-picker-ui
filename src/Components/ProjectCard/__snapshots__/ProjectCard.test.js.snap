// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Card should match the snapshot 1`] = `
<article
  className="project-card__article"
>
  <div
    className="project-card__header"
  >
    <img
      alt="edit icon"
      onClick={[Function]}
      src="editIcon.svg"
    />
    <input
      className="project-edit__input"
      disabled={true}
      maxlength="25"
      onChange={[Function]}
      type="text"
      value="Cool Project"
    />
    <img
      alt="trashIcon"
      onClick={[Function]}
      src="trashIcon.svg"
    />
  </div>
  <PaletteCard
    key="0"
    palette={
      Object {
        "colorFive": "#FFFFFF",
        "colorFour": "#FFFFFF",
        "colorOne": "#FFFFFF",
        "colorThree": "#FFFFFF",
        "colorTwo": "#FFFFFF",
        "id": 1,
        "name": "Cool Colors",
        "project_id": "12345",
      }
    }
    showPalette={[MockFunction]}
    trashPalette={[MockFunction]}
    updatePaletteName={[MockFunction]}
  />
</article>
`;

exports[`Project Card should match the snapshot when the project name is edittable 1`] = `
<article
  className="project-card__article"
>
  <div
    className="project-card__header"
  >
    <img
      alt="save icon"
      onClick={[Function]}
      src="saveIcon.svg"
    />
    <input
      className="project-edit__input"
      disabled={false}
      maxlength="25"
      onChange={[Function]}
      type="text"
      value="Cool Project"
    />
    <img
      alt="trashIcon"
      onClick={[Function]}
      src="trashIcon.svg"
    />
  </div>
  <PaletteCard
    key="0"
    palette={
      Object {
        "colorFive": "#FFFFFF",
        "colorFour": "#FFFFFF",
        "colorOne": "#FFFFFF",
        "colorThree": "#FFFFFF",
        "colorTwo": "#FFFFFF",
        "id": 1,
        "name": "Cool Colors",
        "project_id": "12345",
      }
    }
    showPalette={[MockFunction]}
    trashPalette={[MockFunction]}
    updatePaletteName={[MockFunction]}
  />
</article>
`;
